triggers/service/src/main/scala/com/digitalasset/daml/lf/engine/trigger/ServiceMain.scala

```diff
@@ -96,24 +96,33 @@ object TriggerActor {
         sslContext = None,
       )
 
-      // Waiting for the ACS query to finish so we can build the initial state.
-      // TODO We should handle being stopped while querying the ACS.
-      def queryingACS() = Behaviors.receiveMessagePartial[Message] {
-        case QueryACSFailed(cause) => throw new RuntimeException("ACS query failed", cause)
-        case QueriedACS(runner, acs, offset) =>
-          val (killSwitch, trigger) = runner.runWithACS(
-            acs,
-            offset,
-            msgFlow = KillSwitches.single[TriggerMsg],
-          )
-          // TODO If we are stopped we will end up causing the future to complete which will trigger
-          // a message that is sent to a now terminated actor. We should fix this somehow™.
-          context.pipeToSelf(trigger) {
-            case Success(_) => Failed(new RuntimeException("Trigger exited unexpectedly"))
-            case Failure(cause) => Failed(cause)
-          }
-          running(killSwitch)
-      }
+      // Waiting for the ACS query to finish so we can build the
+      // initial state.
+      def queryingACS(wasStopped: Boolean): Behaviors.Receive[Message] =
+        Behaviors.receiveMessagePartial[Message] {
+          case QueryACSFailed(cause) => throw new RuntimeException("ACS query failed", cause)
+          case QueriedACS(runner, acs, offset) =>
+            val (killSwitch, trigger) = runner.runWithACS(
```

Comment:
This doesn’t look right to me. If we got stopped we don’t want to spawn the trigger here only to then tear it down once we receive the stop message. This should just not start in the first place.

Reply:
Ok. I see what you mean. I'll adjust it accordingly. Thanks!